### PR TITLE
provider/google: set additional_zones to computed and disallow the original zone from appearing in the list

### PR DESCRIPTION
**Pull request by danawillow**
This fixes a bug introduced by #11018.

`cluster.Locations` will never be empty on read (because it will contain at least `cluster.Zone`), so even if `additional_zones` wasn't set in the config, it was getting set on read, causing a subsequent `terraform plan` to be non-empty.

### CODE REVIEW

File: builtin/providers/google/resource_container_cluster.go [hunk 1]
```diff
@@ -444,7 +444,10 @@ func resourceContainerClusterRead(d *schema.ResourceData, meta interface{}) erro
 
 	d.Set("name", cluster.Name)
 	d.Set("zone", cluster.Zone)
-	d.Set("additional_zones", cluster.Locations)
+
+	if _, ok := d.GetOk("additional_zones"); ok {
+		d.Set("additional_zones", cluster.Locations)
```
Comment (drebes) [inline builtin/providers/google/resource_container_cluster.go]: Alternatively, you can set "additional_zones" to be Computed. See #11631 for the same problem there.

Comment (danawillow) [inline builtin/providers/google/resource_container_cluster.go]: Done

**Comment by danawillow** [comment]
I realized that there's no good way of being able to allow the original zone in the `additional_locations` list, so I went ahead and disallowed it, since it's called additional_locations anyway.

Why this was necessary: if I don't include the original zone, then when I get back `locations` from the API it will include it and update the state file. The next update then will show the addition of the original zone as a change. If I proactively remove the original zone from the `locations` list on read, then the opposite issue occurs if I did have the original zone in the list.

**Comment by paddyforan** [comment]
As a general note, I'm in support of getting away from the practice of only setting state if it's set in config. That's causing headaches and is kind of a special-case thing, in my opinion. So glad to see the change to computed. I have some code-level questions, but I'm going to read the code a bit more closely before I pester you with silly questions. Queuing this up for post-lunch. :)

**Comment by paddyforan** [comment]
OK, after reading through all this, all my questions were, in fact, silly. So the last big question I have is it seems the only test changes were to parameterize an existing test without changing the logic of it. Did the test already fail for this? Do we have a test case that shows this bug as failing yet? If not, is it possible to add one?

**Comment by danawillow** [comment]
Yup, the test is already failing. From TeamCity:
```
TestAccContainerCluster_withAdditionalZones 	
 	Current failure:	refs/heads/master	#244992c 	Changes (33) 	07 Feb 17 06:13
	First failure:		#998155f 	Changes (41) 	19 Jan 17 05:09
 
------- Stdout: -------
=== RUN   TestAccContainerCluster_withAdditionalZones
--- FAIL: TestAccContainerCluster_withAdditionalZones (311.81s)
    testing.go:265: Step 0 error: Check failed: Check 2/2 error: number of additional zones did not match 2
FAIL
```

All the other ContainerCluster tests are failing too (which prompted this fix):
```
TestAccContainerCluster_basic 	
 	Current failure:	refs/heads/master	#244992c 	Changes (33) 	07 Feb 17 06:13
	First failure:		#cb8d7bf 	Changes (27) 	12 Jan 17 05:09
 
------- Stdout: -------
=== RUN   TestAccContainerCluster_basic
--- FAIL: TestAccContainerCluster_basic (301.73s)
    testing.go:265: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        DESTROY/CREATE: google_container_cluster.primary
          additional_zones.#:                   "1" => "0" (forces new resource)
          additional_zones.0:                   "*******-a" => "" (forces new resource)
[etc.]
```

The test change was because I thought I was going to have to make a change but then didn't. Happy to take it out of the PR if you don't think it belongs here.

**Comment by paddyforan** [comment]
I think the test change is totally great :) I just wanted to be sure we had a failing test, to test regressions and such. I don't know if you can tell, but I'm trying to make sure every code change has a test associated with it. :)

File: builtin/providers/google/resource_container_cluster.go [hunk 2]
```diff
@@ -444,7 +441,17 @@ func resourceContainerClusterRead(d *schema.ResourceData, meta interface{}) erro
 
 	d.Set("name", cluster.Name)
 	d.Set("zone", cluster.Zone)
-	d.Set("additional_zones", cluster.Locations)
+
+	if len(cluster.Locations) > 1 {
```
Comment (paddyforan) [inline builtin/providers/google/resource_container_cluster.go]: It probably doesn't matter, but I'd love to set this in state, even if it's getting set to an empty value. I think, as a general rule of thumb, things should get set in state unless there's a core bug that makes things not work, we can articulate why diffing on it is undesirable, or we can articulate a strong security concern for storing it in state.

Comment (danawillow) [inline builtin/providers/google/resource_container_cluster.go]: Done.

**Comment by paddyforan** [comment]
Looks great, merge at will! Thanks for tackling those test failures, @danawillow. You're awesome.

**Pull request merged**